src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionKind.cs

```diff
@@ -47,6 +47,11 @@ internal enum ConversionKind : byte
         // IdentityValue is not a part of the language. 
         // It is used by lowering to ensure that trivially reduced expressions 
         // do not become exposed to mutations if used as receivers of struct methods.
-        IdentityValue,  
+        IdentityValue,
+
+        // ObjectToPointer is not directly a part of the language
+        // It is used by lowering of "fixed" statements to represent conversion of an object reference (O) to an unmanaged pointer (*)
+        // The conversion is unsafe and makes sense only if (O) is pinned.
+        ObjectToPointer,
```

Comment:
String does not expose ref to its chars, so we have to convert the string object reference itself to a pointer and then offset by a constant to get to the first char.

Any better suggestions how this could be called?

---
In reply to: [124919088](https://github.com/dotnet/roslyn/pull/20548#discussion_r124919088) [](ancestors = 124919088,124912940)